dpppt-backend-sdk/dpppt-backend-sdk-ws/src/test/java/org/dpppt/backend/sdk/ws/insertmanager/InsertManagerTest.java

```diff
@@ -7,18 +7,75 @@
 import ch.qos.logback.classic.Logger;
 import ch.qos.logback.classic.spi.ILoggingEvent;
 import ch.qos.logback.core.AppenderBase;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
 import java.time.Duration;
+import java.time.Instant;
 import java.util.ArrayList;
+import java.util.Base64;
 import java.util.List;
+import java.util.UUID;
 import org.dpppt.backend.sdk.model.gaen.GaenKey;
 import org.dpppt.backend.sdk.semver.Version;
 import org.dpppt.backend.sdk.utils.UTCInstant;
 import org.dpppt.backend.sdk.ws.insertmanager.insertionmodifier.OldAndroid0RPModifier;
 import org.dpppt.backend.sdk.ws.util.ValidationUtils;
 import org.junit.Test;
 import org.slf4j.LoggerFactory;
+import org.springframework.security.oauth2.jwt.Jwt;
 
 public class InsertManagerTest {
+
+  protected String createToken(boolean fake, UTCInstant expiresAt) {
+    Claims claims = Jwts.claims();
+    claims.put("scope", "exposed");
+    claims.put("onset", "2020-04-20");
+    claims.put("fake", fake ? "1" : "0");
+    return Jwts.builder()
+        .setClaims(claims)
+        .setId(UUID.randomUUID().toString())
+        .setSubject("test-subject" + UTCInstant.now().getOffsetDateTime().toString())
+        .setExpiration(expiresAt.getDate())
+        .setIssuedAt(UTCInstant.now().getDate())
+        .compact();
+  }
+
+  @Test
+  public void fakeJWTDoesNotInsert() throws Exception {
+    InsertManager manager =
+        new InsertManager(
+            new MockDataSource(),
+            new ValidationUtils(16, Duration.ofDays(14), Duration.ofHours(2).toMillis()));
+    var jwt =
+        Jwt.withTokenValue(createToken(true, UTCInstant.now().plusMinutes(5)))
+            .header("alg", "HS256")
+            .claim("fake", "1")
+            .build();
+
+    var gaenKey1 = new GaenKey();
+    gaenKey1.setKeyData(Base64.getEncoder().encodeToString("testKey32Bytes--".getBytes("UTF-8")));
+    gaenKey1.setRollingStartNumber(
+        (int) Duration.ofMillis(Instant.now().toEpochMilli()).dividedBy(Duration.ofMinutes(10)));
+    gaenKey1.setRollingPeriod(144);
+    gaenKey1.setTransmissionRiskLevel(0);
+    var gaenKey2 = new GaenKey();
+    gaenKey2.setRollingStartNumber(
+        (int)
+            Duration.ofMillis(Instant.now().minus(Duration.ofDays(1)).toEpochMilli())
+                .dividedBy(Duration.ofMinutes(10)));
+    gaenKey2.setKeyData(Base64.getEncoder().encodeToString("testKey32Bytes--".getBytes("UTF-8")));
+    gaenKey2.setRollingPeriod(144);
+    gaenKey2.setTransmissionRiskLevel(0);
+
+    gaenKey1.setFake(0);
+    gaenKey2.setFake(0);
+    List<GaenKey> exposedKeys = new ArrayList<>();
+    exposedKeys.add(gaenKey1);
+    exposedKeys.add(gaenKey2);
+
+    manager.insertIntoDatabase(exposedKeys, "test", jwt, UTCInstant.now());
+  }
```

Comment:
There doesn't seem to be an actual 'test' here - no `assertEqual` or such... am I missing something?

Also, like with the test above, I usually prefer to do the double-testing in the same test:
- test it actually gets inserted with `fake == 0`
- test it gets refused with `fake == 0`

Because I think we got the case where a refusal to store a bad key was due to a previous error, but not the check the test was supposed to do...

Reply:
> There doesn't seem to be an actual 'test' here - no assertEqual or such... am I missing something? 

It uses a `MockDataSource` which throws a `RuntimeException` if there is a call to `UpsertExposees`. 

The idea of this test is, to ONLY test the `InsertManager` totally DataProvider agnostic. 

If I'm not mistaken, we check that insertion into the database woks correctly in various integration tests. Here we only ensure that nothing is inserted if there is a JWT with `fake=1` (or rather `fake != 0`).

Reply:
but yeah to show that it indeed passes the filters (also if more are added) would be preferable. @UBaggeler same here? Issue?